### DISCUSSION TITLE
SUPABASE_URLとSUPABASE_ANON_KEYの設定について

### DISCUSSION THREAD

**Discussion by taro-aoyama** [flutter-osaka-dev/flutter-chat#8] [Q&A] [answered]
イベント開催していただいてありがとうございました。
Flutter初学者なのでとても勉強になりました。
また、Supabaseもはじめて触ったのですがとても良いものな気がしました。
今後どんどん使わせてもらおうと思います。

１点質問があります。

```
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();

  await Supabase.initialize(
    // TODO: ここにSupabaseのURLとAnon Keyを入力
    url: 'SUPABASE_URL',
    anonKey: 'SUPABASE_ANON_KEY',
  );
  runApp(const MyApp());
}
```

ハンズオンでのサンプルでは、main.dartに
SUPABASE_URLとSUPABASE_ANON_KEYを直接記載する例となっていましたが、
このままリポジトリにコミットするとちょっと怖いなぁと思いました。
外部ファイルに書いて読み込んだりとか、見えないようにすることはできますか？？

**Comment by dshukertjr** [comment, marked as answer]
ご参加kありがとうございました！

はい、Flutterでも.envファイルを使って各種環境変数を管理することが可能となっております！

こちらのパッケージを使うと.envファイルの環境数変数を読み込むことができ、環境変数をコミットせずに管理することができます！
https://pub.dev/packages/flutter_dotenv
